tests/test_loss_masking.py

```diff
@@ -0,0 +1,70 @@
+from unittest.mock import MagicMock
+
+import pytest
+import torch
+
+from modalities.batch import DatasetBatch
+from modalities.models.components.collator_wrapper import (
+    LossMaskingCollateFnWrapper,
+    LossMaskingCollateFnWrapperConfig,
+    MaskingTokenConfig,
+)
+from modalities.models.gpt2.collator import GPT2LLMCollateFn
+from modalities.tokenization.tokenizer_wrapper import TokenizerWrapper
+
+
+def tokenize(word: str):
+    vocab = {"begin": 0, "end": 1}
+    return vocab[word]
+
+
+@pytest.fixture
+def dummy_tokenizer():
+    mock = MagicMock(spec=TokenizerWrapper)
+    mock.get_token_id.side_effect = tokenize
+    return mock
+
+
+@pytest.fixture
+def loss_masking_config(dummy_tokenizer) -> LossMaskingCollateFnWrapperConfig:
+    return dict(
+        collate_fn=GPT2LLMCollateFn(sample_key="sample", target_key="target"),
+        target_keys_to_mask=["target"],
+        loss_ignore_index=-100,
+        mask_tokens=MaskingTokenConfig(b_include_to_loss_token="begin", e_include_to_loss_token="end"),
+        tokenizer=dummy_tokenizer,
+    )
+
+
+# calculating nce_loss for two randomly generated batch of embeddings (manually calculated)
+@pytest.mark.parametrize(
+    "batch,expected_batch",
+    [
+        (
+            [
+                # the collate_fn will shift the sample and target:
+                # shifted sample:           [5, 5, 0, 5, 5, 1, 5, 0, 5, 1, 0, 1, 5, 0]
```

Comment:
is the sample really "shifted"? We just cut off the last element no?

Reply:
Right, the sample is cut at the end, and the target is at the beginning. I will change the name